### DISCUSSION TITLE
Updating material property name from "thermal_conductivity" to "thermal_conductivity_tensor"

### DISCUSSION THREAD

**Discussion by singhgp4321** [idaholab/moose#19096] [Q&A General] [answered]
Does anyone mind if I update the declared and extracted material property name (type RankTwoTensor) from "thermal_conductivity" to "thermal_conductivity_tensor" in the `AnisoHeatConductionMaterial` and `AnisoHeatConduction` kernel?

@dschwen let me know if you are fine with the following change?

The following code in `AnisoHeatConductionMaterial`

    _thermal_conductivity(
        declareGenericProperty<RankTwoTensor, is_ad>(_base_name + "thermal_conductivity")),
    _dthermal_conductivity_dT(
        declarePropertyDerivative<RankTwoTensor>(_base_name + "thermal_conductivity", _T_name)),

and in `AnisoHeatConduction` kernel

    _k(getMaterialProperty<RankTwoTensor>("thermal_conductivity")),
    _dk_dT(getMaterialPropertyDerivative<RankTwoTensor>("thermal_conductivity", _var.name()))

_will change to:_

    _thermal_conductivity(
        declareGenericProperty<RankTwoTensor, is_ad>(_base_name + "thermal_conductivity_tensor")),
    _dthermal_conductivity_dT(
        declarePropertyDerivative<RankTwoTensor>(_base_name + "thermal_conductivity_tensor", _T_name)),

    _k(getMaterialProperty<RankTwoTensor>("thermal_conductivity_tensor")),
    _dk_dT(getMaterialPropertyDerivative<RankTwoTensor>("thermal_conductivity_tensor", _var.name()))

**Comment by hugary1995** [comment, marked as answer]
That's gonna be inconsistent with the existing naming convention IMHO. For example, we have "stress" instead of "stress_tensor", "eigenstrain_names" instead of "eigenstrain_tensor_names". Is the original name "thermal_conductivity" causing you trouble? If so, I'm fine with renaming it.
